### PR TITLE
Feature: add select component

### PR DESCRIPTION
**Pull request by baomastr**

### CODE REVIEW

**Review by therealemjy — CHANGES_REQUESTED** [review]
Hey,
Sorry but this doesn't match the designs at all:
Screen Shot 2022-05-05 at 17 06 38
Designs:
Screen Shot 2022-05-05 at 17 06 32

**Comment by therealemjy** [comment]
Much better 👍

File: src/components/v2/Select/index.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  options: Option[];
+}
+
+const DEFAULT_OPTION = {
+  value: '',
+  label: 'All',
+};
+
+export const Select = ({ className, options }: ISelectProps) => {
+  const [selectedValue, setSelectedValue] = React.useState(DEFAULT_OPTION.value);
```
Comment (therealemjy) [inline src/components/v2/Select/index.tsx]: The state and its modifier function should be passed through props, as we'll always need to use the state from the parent component that renders it

File: src/components/v2/Select/index.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+interface ISelectProps {
+  className?: string;
+  options: Option[];
+}
+
+const DEFAULT_OPTION = {
+  value: '',
+  label: 'All',
+};
```
Comment (therealemjy) [inline src/components/v2/Select/index.tsx]: I'd remove that to make it more flexible (not all selects will have a default "All" option).

File: src/components/v2/Select/index.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+export const Select = ({ className, options }: ISelectProps) => {
```
Comment (therealemjy) [inline src/components/v2/Select/index.tsx]: Please use `React.FC` to type it like we do with other components

**Review by therealemjy — COMMENTED** [review]
It displays a title on mobile:
Screen Shot 2022-05-08 at 22 05 57

Since we don't have mobile designs for this component I think it's fine to use MUI's default one, but let's remove the title.

File: src/components/v2/Select/index.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { withCenterStory } from 'stories/decorators';
+import { Select } from '.';
+
+export default {
+  title: 'Components/Select',
+  component: Select,
+  decorators: [withCenterStory({ width: 200 })],
+  parameters: {
+    backgrounds: {
+      default: '#1F2028',
```
Comment (coreyar) [inline src/components/v2/Select/index.stories.tsx]: Ideally this value comes from the enum of options we have set `Primary` `Paper` `White`. They are set in `.storybook/preview.js` if not we can grab them from the Palette constant

**Comment by baomastr** [comment]
@therealemjy 

> Since we don't have mobile designs for this component I think it's fine to use MUI's default one, but let's remove the title.

here are mobile designs:

image

**Comment by therealemjy** [comment]
Ah right, nice 👍
Looks good to me.
@coreyar Thanks for updating the PR

**Review by therealemjy — APPROVED** [review]

File: src/components/v2/Select/index.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      anchorOrigin: {
+        vertical: 'top',
+        horizontal: 'center',
+      },
+    };
+    return {
+      PaperProps: {
+        sx: styles.menuWrapper,
+      },
+      ...(isSmDown ? mobileStyles : {}),
```
Comment (therealemjy) [inline src/components/v2/Select/index.tsx:63]: Actually, I just saw that.

We want to move away from using JS for responsiveness: let's use CSS directly here (you can render both mobile and desktop instances of the component and choose which one to render based on the screen size using CSS directly).
Also, please don't copy and paste the whole code: you can make it so only the style prop differs from one version to the other.

**Review by therealemjy — CHANGES_REQUESTED** [review]

**Comment by therealemjy** [comment]
Actually never mind, we can't use CSS only here because of how MUI renders the dropdown menu.
Looks all good to me then 👍

**Review by therealemjy — APPROVED** [review]

**Pull request merged**